lib/core/src/sync/model/mod.rs

```diff
@@ -1,51 +1,184 @@
+use std::sync::Arc;
+
+use lwk_wollet::hashes::hex::DisplayHex as _;
 use serde::{Deserialize, Serialize};
 
-use crate::model::{Direction, InternalSwapTree};
+use self::sync::{ListChangesRequest, ListenChangesRequest, Record, SetRecordRequest};
+use crate::model::{
+    ChainSwap, Direction, PaymentState, ReceiveSwap, SendSwap, Signer, SignerError,
+};
+
+use super::CURRENT_SCHEMA_VERSION;
 
 pub(crate) mod sync;
 
 #[derive(Serialize, Deserialize)]
 pub(crate) struct ChainSyncData {
     pub(crate) swap_id: String,
     pub(crate) preimage: String,
-    pub(crate) description: Option<String>,
+    pub(crate) create_response_json: String,
     pub(crate) direction: Direction,
-    pub(crate) claim_swap_tree: InternalSwapTree,
-    pub(crate) claim_fees_sat: u64,
+    pub(crate) lockup_address: String,
     pub(crate) claim_address: String,
+    pub(crate) claim_fees_sat: u64,
     pub(crate) claim_private_key: String,
-    pub(crate) claim_timeout_block_height: u32,
-    pub(crate) lockup_swap_tree: InternalSwapTree,
-    pub(crate) lockup_address: String,
     pub(crate) refund_private_key: String,
-    pub(crate) lockup_timeout_block_height: u32,
+    pub(crate) timeout_block_height: u32,
     pub(crate) payer_amount_sat: u64,
     pub(crate) receiver_amount_sat: u64,
     pub(crate) accept_zero_conf: bool,
+    pub(crate) created_at: u32,
+    pub(crate) description: Option<String>,
+}
+
+impl From<ChainSwap> for ChainSyncData {
+    fn from(value: ChainSwap) -> Self {
+        Self {
+            swap_id: value.id,
+            preimage: value.preimage,
+            create_response_json: value.create_response_json,
+            direction: value.direction,
+            lockup_address: value.lockup_address,
+            claim_address: value.claim_address,
+            claim_fees_sat: value.claim_fees_sat,
+            claim_private_key: value.claim_private_key,
+            refund_private_key: value.refund_private_key,
+            timeout_block_height: value.timeout_block_height,
+            payer_amount_sat: value.payer_amount_sat,
+            receiver_amount_sat: value.receiver_amount_sat,
+            accept_zero_conf: value.accept_zero_conf,
+            created_at: value.created_at,
+            description: value.description,
```

Comment:
What do you think about separating the synced data into a different table for each swap? 
We will need to add join for each query but perhaps it makes it more clear and separated what persistent data we sync/udpate and what data is in responsibility of the client to update locally?

Reply:
To your point, I think it definitely helps making the database structure have more sense (we can look at the schema and have a better idea of how the implementation is going to look like).

Is this necessary from a code standpoint, though? Taking a look at the sync models IMO is enough to do what you said: distinguish which fields will be synced and which won't. Since these models are only necessary for transport (meaning that, in the end, we still have to convert to a swap to insert to the tables), I don't see the need to reflect these abstractions to the database.